src/visGeometry/VolumeModel.ts

```diff
@@ -1,40 +1,146 @@
+            if (channel < numChannels) {
+                this.channelsEnabled[channel] = true;
+            }
+        }
+
+        for (const [channelIndex, enabled] of this.channelsEnabled.entries()) {
+            this.drawable.setVolumeChannelEnabled(channelIndex, enabled);
+        }
+    }
 
     public setImage(volumeObject: Volume): void {
-        this.image = new VolumeDrawable(volumeObject, {});
+        this.volume = volumeObject;
+        this.drawable = new VolumeDrawable(this.volume, {});
+        this.volume.addVolumeDataObserver(this);
+        this.drawable.setBrightness(0.7);
+        this.drawable.setGamma(0.15, 0.9, 1.0);
+        this.drawable.setDensity(0.7);
     }
 
     public setAgentData(data: AgentData): void {
-        if (this.image) {
-            this.image.setTranslation(new Vector3(data.x, data.y, data.z));
-            this.image.setRotation(new Euler(data.xrot, data.yrot, data.zrot));
-            const r = data.cr * 2;
-            this.image.setScale(new Vector3(r, r, r));
+        if (this.drawable) {
+            this.drawable.setTranslation(new Vector3(data.x, data.y, data.z));
+            this.drawable.setRotation(
+                new Euler(data.xrot, data.yrot, data.zrot)
+            );
+            this.scale = data.cr * 2;
+            this.drawable.setScale(
+                new Vector3(this.scale, this.scale, this.scale)
+            );
+            // Always defined if `drawable` is, but ts doesn't know that.
+            if (this.volume) {
+                // Volume agent data may use subpoint 0 as time
+                const numPoints = data.subpoints.length;
+                const time = numPoints > 0 ? data.subpoints[0] : 0;
+                if (this.volume.loadSpec.time !== time) {
+                    this.volume.updateRequiredData({ time });
+                }
+                // If there are more subpoints, they are enabled channel idxes.
+                // Otherwise, just channel 0 is enabled.
+                const channels = numPoints > 1 ? data.subpoints.slice(1) : [0];
+                this.setEnabledChannels(channels);
+            }
         }
     }
 
+    public loadInitialData(): void {
+        this.volume?.loader?.loadVolumeData(this.volume);
+    }
+
     public getObject3D(): Object3D | undefined {
-        return this.image?.sceneRoot;
+        return this.drawable?.sceneRoot;
+    }
+
+    public onChannelLoaded(_vol: Volume, channelIndex: number): void {
+        if (this.drawable) {
+            const isEnabled = this.channelsEnabled[channelIndex];
+            this.drawable.setVolumeChannelEnabled(channelIndex, isEnabled);
+            this.drawable.updateScale();
+            this.drawable.onChannelLoaded([channelIndex]);
+            if (this.volume) {
+                const histo = this.volume.getHistogram(channelIndex);
+                const min = histo.findBinOfPercentile(0.5);
+                const max = histo.findBinOfPercentile(0.983);
```

Comment:
I saw the note you had in the PR description, maybe tag all the hardcoded values with a TODO so they're easier to track down? Totally optional